### PR TITLE
beanshell methodselector applied at suite level is ignored

### LINKED ISSUE #1674: beanshell methodselector applied at suite level is ignored
6.13.1

### Expected behavior
beanshell method selector applied at suite level should be used.

### Actual behavior
beanshell method selector applied at suite level gets ignored for the tests.

### Is the issue reproductible on runner?

- [X] Shell
- [X] Maven
- [ ] Gradle
- [ ] Ant
- [ ] Eclipse
- [X] IntelliJ
- [ ] NetBeans

```java
import org.testng.annotations.Test;

public class SampleTestclass {
    @Test
    public void testMethod() {
        System.err.println("Hello World");
    }
}
```

suite xml

```xml
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE suite SYSTEM "http://testng.org/testng-1.0.dtd">
<suite name="1265_Suite" parallel="false" verbose="2">
    <method-selectors>
        <method-selector>
            <script language="beanshell"><![CDATA[
                    System.out.println("Executing the test [" + testngMethod.getXmlTest().getName() + "]");
                    return true;
                ]]></script>
        </method-selector>
    </method-selectors>
    <test name="rajni">
        <classes>
            <class name="com.rationaleemotions.github.beanshellproblem.SampleTestclass"/>
        </classes>
    </test>
    <test name="kamal">
        <classes>
            <class name="com.rationaleemotions.github.beanshellproblem.SampleTestclass"/>
        </classes>
    </test>
    <test name="mgr">
        <classes>
            <class name="com.rationaleemotions.github.beanshellproblem.SampleTestclass"/>
        </classes>
    </test>
</suite>
```

output

```
objc[11335]: Class JavaLaunchHelper is implemented in both /Library/Java/JavaVirtualMachines/jdk1.8.0_131.jdk/Contents/Home/bin/java (0x10eb2c4c0) and /Library/Java/JavaVirtualMachines/jdk1.8.0_131.jdk/Contents/Home/jre/lib/libinstrument.dylib (0x10ebb84e0). One of the two will be used. Which one is undefined.
...
... TestNG 6.14.0-RC1 by Cédric Beust (cedric@beust.com)
...
Hello World
Hello World
PASSED: testMethod

===============================================
    rajni
    Tests run: 1, Failures: 0, Skips: 0
===============================================
PASSED: testMethod

===============================================
    kamal
    Tests run: 1, Failures: 0, Skips: 0
===============================================
Hello World
PASSED: testMethod

===============================================
    mgr
    Tests run: 1, Failures: 0, Skips: 0
===============================================

===============================================
1265_Suite
Total tests run: 3, Failures: 0, Skips: 0
===============================================


Process finished with exit code 0
```

### PR DESCRIPTION
**Pull request by krmahadevan**
Closes #1674

Fixes #1674 .

### Did you remember to?

- [X] Add test case(s)
- [X] Update `CHANGES.txt`

We encourage pull requests that:

* Add new features to TestNG (or)
* Fix bugs in TestNG

If your pull request involves fixing SonarQube issues then we would suggest that you please discuss this with the 
[TestNG-dev](https://groups.google.com/forum/#!forum/testng-dev) before you spend time working on it.

### CODE REVIEW

File: src/test/java/org/testng/xml/XmlSuiteTest.java [hunk 1]
```diff
@@ -71,4 +71,17 @@ public void ensureNoExceptionsAreRaisedWhenMethodSelectorsDefinedAtSuiteLevel()
...
+    public void ensureSuiteLevelBeanshellIsAppliedToAllTests() throws IOException {
+        Parser parser = new Parser("src/test/resources/xml/issue1674.xml");
+        List<XmlSuite> suites = parser.parseToList();
+        XmlSuite xmlsuite = suites.get(0);
+        assertThat(xmlsuite.getTests().get(0).getMethodSelectors().size()).isEqualTo(0);
+        TestNG testNG = create();
+        testNG.setXmlSuites(suites);
+        testNG.setUseDefaultListeners(false);
+        testNG.run();
+        assertThat(xmlsuite.getTests().get(0).getMethodSelectors().size()).isEqualTo(1);
```
Comment (cbeust) [inline src/test/java/org/testng/xml/XmlSuiteTest.java]: Is there any way to assert that the BeanShell code actually ran instead of just testing this?

Comment (krmahadevan) [reply src/test/java/org/testng/xml/XmlSuiteTest.java]: @cbeust - I added an assertion that will check to see if the beanshell got executed. Can you please check if thats fine ?

Comment (cbeust) [reply src/test/java/org/testng/xml/XmlSuiteTest.java]: Looks good to me, feel free to merge if you're happy with it.

Thanks!

File: src/main/java/org/testng/TestNG.java [hunk 2]
```diff
@@ -910,6 +916,7 @@ private void initializeConfiguration() {
       //
       for (XmlMethodSelector methodSelector : s.getMethodSelectors() ) {
         addMethodSelector(methodSelector.getClassName(), methodSelector.getPriority());
```
Comment (juherr) [inline src/main/java/org/testng/TestNG.java]: I don't catch why we need the selector in both m_methodDescriptors and m_selectors. I think we will have selector duplication after https://github.com/cbeust/testng/pull/1675/files#diff-c89ea27c996c668601a33f887b5675f2R1270
Why do we want it?

Comment (krmahadevan) [reply src/main/java/org/testng/TestNG.java]: @juherr - They both are different from what I have seen in the codebase. `m_methodDescriptors` basically captures only those method selectors for which a classname and priority has been defined. So I believe these are limited to `org.testng.IMethodSelector` implementations that a end-user can wire in. `m_selector` is basically for beanshells, which TestNG injects from within, whenever it sees a beanshell expression.

**Pull request merged**